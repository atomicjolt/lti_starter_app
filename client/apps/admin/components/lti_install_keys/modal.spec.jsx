import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Modal from './modal';

const mockStore = configureStore([]);
const store = mockStore({});

describe('lti install key modal', () => {

  let result;
  let props;
  let closed;
  let saved;
  const name = 'the application';

  beforeEach(() => {
    saved = false;
    closed = false;
    props = {
      isOpen: true,
      closeModal: () => { closed = true; },
      save: () => { saved = true; },
      ltiInstallKey: {
        id: 2,
        clientId: 'lti-key',
        iss: 'iss',
        jwksUrl: 'jwksUrl',
        tokenUrl: 'tokenUrl',
        oidcUrl: 'oidcUrl',
        created_at: 'created_at',
      },
      application: {
        id: 1,
        name
      }
    };
    result = TestRenderer.create(
      <Provider store={store}>
        <Modal {...props} />
      </Provider>
    );
  });

  it('handles the save function', () => {
    expect(saved).toBeFalsy();
    result.root.save();
    expect(saved).toBeTruthy();
  });

  it('handles the close function', () => {
    expect(closed).toBeFalsy();
    result.root.closeModal();
    expect(closed).toBeTruthy();
  });
});
