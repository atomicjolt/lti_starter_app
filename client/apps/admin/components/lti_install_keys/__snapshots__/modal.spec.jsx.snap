// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lti install key modal matches the snapshot 1`] = `
<div
  className="ReactModal__Overlay c-modal__background"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={Object {}}
>
  <div
    aria-label="Lti Install Key Modal"
    aria-modal={true}
    className="ReactModal__Content c-modal c-modal--settings is-open"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex="-1"
  >
    <h2
      className="c-modal__title"
    >
      Update
       
      Lti Install Key
    </h2>
    <form>
      <div
        className="o-grid o-grid__modal-top"
      >
        <div
          className="o-grid__item u-half"
        >
          <div
            className="c-input--container"
          >
            <label
              className="c-input"
              htmlFor="iss_input"
            >
              <span>
                iss
              </span>
              <input
                id="iss_input"
                name="iss"
                onChange={[Function]}
                type="text"
                value="iss"
              />
            </label>
          </div>
        </div>
        <div
          className="o-grid__item u-half"
        >
          <div
            className="c-input--container"
          >
            <label
              className="c-input"
              htmlFor="client_id_input"
            >
              <span>
                client id
              </span>
              <input
                id="client_id_input"
                name="clientId"
                onChange={[Function]}
                type="text"
                value="lti-key"
              />
            </label>
          </div>
        </div>
        <div
          className="o-grid__item u-half"
        >
          <div
            className="c-input--container"
          >
            <label
              className="c-input"
              htmlFor="jwks_url_input"
            >
              <span>
                jwks url
              </span>
              <input
                id="jwks_url_input"
                name="jwksUrl"
                onChange={[Function]}
                type="text"
                value="jwksUrl"
              />
            </label>
          </div>
        </div>
        <div
          className="o-grid__item u-half"
        >
          <div
            className="c-input--container"
          >
            <label
              className="c-input"
              htmlFor="token_url_input"
            >
              <span>
                token url
              </span>
              <input
                id="token_url_input"
                name="tokenUrl"
                onChange={[Function]}
                type="text"
                value="tokenUrl"
              />
            </label>
          </div>
        </div>
        <div
          className="o-grid__item u-half"
        >
          <div
            className="c-input--container"
          >
            <label
              className="c-input"
              htmlFor="oidc_url_input"
            >
              <span>
                oidc url
              </span>
              <input
                id="oidc_url_input"
                name="oidcUrl"
                onChange={[Function]}
                type="text"
                value="oidcUrl"
              />
            </label>
          </div>
        </div>
      </div>
      <button
        className="c-btn c-btn--yellow"
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
      <button
        className="c-btn c-btn--gray--large u-m-right"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </form>
  </div>
</div>
`;
